### PR TITLE
Issue #28: As a DevOps, I would like to automate Azure infrastructure deployment using Terraform

### PR DESCRIPTION
**Pull request by SonOfLope**
This pull request introduces the necessary configurations on the testing repository from our Azure Devops Repository [AI-LAB](https://dev.azure.com/CFIA-DevOps-ACIA/AI-Lab) to establish an integration between our Azure infrastructure and our source control management on GitHub. By using an Azure Service Principal and a GitHub service connector previously configured on our Azure Devops, we are able to execute Terraform workflows within the Azure DevOps AI-LAB repository while using this repository (infra) as source code for the workflows.

I suggest we use this solution as of now to migrate our services to GCP to Azure while I open a ticket with the Cloud team to provide a new service principal providing permissions to this repository (infra) to execute directly to our Azure subscription without the use of the middle-man (AI-LAB repository). 

The following code simply creates a ressource group as an example to prove that the workflow works : 
* Azure test resource group :
Azure test resource group
* Successful pipeline : 
image

This allows us to move on to the next step of migrating our services to our Azure subscription.

TODO 
- [ ] Change reusable workflow Terraform linting reference to main branch once [this](https://github.com/ai-cfia/github-workflows/pull/88) gets merged.

### CODE REVIEW

**Comment by SonOfLope** [comment]
Set in draft to discuss about trigger steps for this workflow

**Comment by rngadam** [comment]
do we want to add https://github.com/ai-cfia/github-workflows/blob/main/.github/workflows/workflow-yaml-check.md and a terraform linter?

**Comment by SonOfLope** [comment]
> do we want to add https://github.com/ai-cfia/github-workflows/blob/main/.github/workflows/workflow-yaml-check.md and a terraform linter?

Added

**Comment by SonOfLope** [comment]
I also added an approval process for terraform apply step. It waits for an approval on Azure Devops AI-Lab repository (where we got access to the Azure service connector and where the Azure pipeline terraform-apply.yml is being applied from). Once one of the listed approvers (Thomas, Ricky or me) accepts the Terraform plan, the job to apply the plan gets executed. Once I document this process I will set the PR ready for review

**Review by ThomasCardin — APPROVED** [review]
LGTM!

File: .tflint.hcl [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    enabled = true
+    version = "0.29.0"
+    source  = "github.com/terraform-linters/tflint-ruleset-aws"
+}
+
+plugin "google" {
+    enabled = true
+    version = "0.26.0"
+    source  = "github.com/terraform-linters/tflint-ruleset-google"
+}
```
Comment (rngadam) [inline .tflint.hcl:22]: EOF newline missing

**Review by rngadam — APPROVED** [review]
approved minus the missing newline EOF (which should have been caught by a github workflow)

**Comment by SonOfLope** [comment]
> approved minus the missing newline EOF (which should have been caught by a github workflow)

I dont think we lint hcl files in our workflows.

**Review by ThomasCardin — APPROVED** [review]
Like jonathan said, we don't lint .hcl file, maybe we could use https://github.com/super-linter/super-linter. I'll write an issue about it.

**Comment by rngadam** [comment]
@ThomasCardin @SonOfLope 

...this is part of the generic standards validation workflow that should apply to ALL files:

https://github.com/ai-cfia/github-workflows/blob/8521ad203de073af433485a150dc00597ca0380b/.github/workflows/workflow-repo-standards-validation.yml#L174

I see it seems to be running, so is this broken or misconfigured?

**Pull request merged**